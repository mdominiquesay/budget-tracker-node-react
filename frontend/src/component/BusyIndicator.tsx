import React from 'react';

const BusyIndicator: React.FC = () => {
  return (
    <div className="busy-indicator">
      <div className="spinner"></div>
      <p>Loading...</p>
    </div>
  );
};

export default BusyIndicator;
