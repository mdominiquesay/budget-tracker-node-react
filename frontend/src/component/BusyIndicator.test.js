// __tests__/BusyIndicator.test.js

import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import BusyIndicator from './BusyIndicator'; // Update the import path as necessary

describe('BusyIndicator', () => {
    it('renders correctly', () => {
        const component = renderer.create(
            <BusyIndicator />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
