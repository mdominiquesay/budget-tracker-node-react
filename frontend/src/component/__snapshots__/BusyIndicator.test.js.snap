// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusyIndicator renders correctly 1`] = `
<div
  className="busy-indicator"
>
  <div
    className="spinner"
  />
  <p>
    Loading...
  </p>
</div>
`;

exports[`BusyIndicator renders correctly 2`] = `
<div
  className="busy-indicator"
>
  <div
    className="spinner"
  />
  <p>
    Loading...
  </p>
</div>
`;
